packages/landing/src/common/components/LineCharts/index.js

```diff
@@ -29,53 +32,31 @@ const LineCharts = (props) => {
       return;
     }
 
-    // Calculate day section
-    const beginDay = Math.floor(moment.duration(projectVotes[0].timestamp).asDays());
-    const endDay = Math.floor(moment.duration(new Date()).asDays());
-
-    let daySection = Math.ceil((endDay - beginDay) / X_NUM);
-    daySection = daySection > 1 ? daySection : 1;
-
-    // Calculate chart datas
-    const dayDatas = [];
-    for (let i = 0; i < X_NUM; i += 1) {
-      const day = i * daySection;
-      dayDatas.push({
-        day,
-        amount: (day - 1) > (endDay - beginDay) ? null : 0,
-      });
-    }
+    const dayDatas = []; // List of data({ x: '2020-01-01', y: 40})
 
+		// Calculate dayDatas
     _.each(projectVotes, (vote) => {
-      const day = Math.floor(moment.duration(vote.timestamp).asDays());
-      const dayDiff = day - beginDay;
-      const index = Math.floor(dayDiff / daySection);
-      console.log('index: ', index);
-      dayDatas[index].amount += vote.amount;
-    });
-
-    for (let i = 1; i < X_NUM; i += 1) {
-      if ((dayDatas[i].day - 1) > (endDay - beginDay)) {
-        continue;
+      const momentFormat = moment(vote.timestamp).format('YYYY-MM-DD');
+      let lastData = _.last(dayDatas);
+
+      if (!lastData || lastData.x != momentFormat) {
+        const data = {
+          x: momentFormat,
+          y: _.isEmpty(dayDatas) ? 0: lastData.y,
+        };
+        dayDatas.push(data);
+        lastData = data;
       }
-      dayDatas[i].amount += dayDatas[i-1].amount;
-    }
 
-    // Set chart datas
-    const labels = _.map(dayDatas, (item) => {
-      return moment(projectVotes[0].timestamp).add(item.day, 'day').format('YYYY/M/D');
-    });
-
-    const amounts = _.map(dayDatas, (item) => {
-      return item.amount;
+      lastData.y += vote.amount;
```

Comment:
1. Use _.groupBy(collection, [iteratee=_.identity]) to group all data by dates.
```
 _.groupBy(projectVotes, (vote) => {
  return moment(vote.timestamp).format('YYYY-MM-DD');
})
```

2. Loop over each group, and sum up all vote.amount, and change it to data.y.